tests/api/linearalgebra/linearalgebra.cpp

```diff
@@ -1123,70 +1119,80 @@ void test_multiply_matrix_matrix_row_major(LinearAlgebra &linear_algebra, int le
     EncryptedMatrix ct_a_transpose = linear_algebra.encrypt_matrix(matrix_a_transpose, unit);
     EncryptedMatrix ct_b = linear_algebra.encrypt_matrix(matrix_b, unit, ct_a_transpose.he_level() - 1);
     EncryptedMatrix ct_c_times_A_times_B = linear_algebra.multiply_row_major(ct_a_transpose, ct_b, scalar);
-    Matrix actual_output = linear_algebra.decrypt(ct_c_times_A_times_B);
 
-    // Transpose of A^T is A
-    Matrix matrix_a = trans(matrix_a_transpose);
-    Matrix expected_output = scalar * prec_prod(matrix_a, matrix_b);
+    if (test) {
```

Comment:
When would this be `false` as this is in the `tests/` directory?

Reply:
No, it's a little more subtle than that. When using multiple evaluators (which this test is doing), you need to have the "circuit" isolated from the surrounding code, because not all functions work for all evaluators. In this test, I need to compute the rotations in the circuit (using `RotationSet`) prior to instantiating the `HomomorphicEval` evaluator. The `RotationSet` evaluator doesn't support decryption (it's a dummy ciphertext, remember?), so the call to `decrypt` can't happen when I evaluate this function with the `RotationSet` evaluator. Similarly, the GTest assertions will fail when I use the `RotationSet` evaluator because no work is being done. The circuit (and its tests) are evaluated many times with different parameters, so this guard ensures that decryption and GTest assertions are only evaluated when it is being run with the Homomorphic Evaluator.